clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -7147,7 +7147,9 @@ void Sema::ProcessDeclAttributeList(
   // good to have a way to specify "these attributes must appear as a group",
   // for these. Additionally, it would be good to have a way to specify "these
   // attribute must never appear as a group" for attributes like cold and hot.
-  if (!D->hasAttr<OpenCLKernelAttr>()) {
+  const FunctionType *FnTy = D->getFunctionType();
+  if (!D->hasAttr<OpenCLKernelAttr>() && FnTy &&
+      FnTy->getCallConv() != CallingConv::CC_AMDGPUKernelCall) {
```

Comment:
Is there no other representation before going straight to the calling conv? Also probably should have an isKernel helper somewhere

Reply:
Nope, I originally thought I could check for `AMDGPUKernelCall` attributes, but those don't get attached. It's only on the return type calling conv.